src/main/java/ai/knowly/langtorch/example/SimpleChatBotWithExplicitKey.java

```diff
@@ -4,44 +4,21 @@
 
 import ai.knowly.langtorch.capability.integration.openai.SimpleChatCapability;
 import ai.knowly.langtorch.hub.LangtorchHub;
+import ai.knowly.langtorch.hub.LangtorchHubModuleRegistry;
 import ai.knowly.langtorch.hub.module.token.TokenUsage;
-import ai.knowly.langtorch.hub.schema.LangtorchHubConfig;
 import ai.knowly.langtorch.hub.schema.OpenAIKeyConfig;
-import ai.knowly.langtorch.processor.openai.chat.OpenAIChatProcessorConfig;
-import ai.knowly.langtorch.store.memory.conversation.ConversationMemory;
-import com.google.common.collect.ImmutableList;
 import com.google.common.flogger.FluentLogger;
-import com.google.inject.AbstractModule;
-import com.google.inject.Provides;
 import java.io.IOException;
 
 public class SimpleChatBotWithExplicitKey {
   private static final FluentLogger logger = FluentLogger.forEnclosingClass();
 
   public static void main(String[] args) throws IOException {
     String openAIKey = "YOUR_OPENAI_API_KEY";
-
-    LangtorchHub langtorchHub =
-        new LangtorchHub(
-            LangtorchHubConfig.builder()
-                .setOpenAIKeyConfig(
-                    OpenAIKeyConfig.builder()
-                        .setReadFromEnvFile(false)
-                        .setOpenAiApiKey(openAIKey)
-                        .build())
-                .build(),
-            ImmutableList.of(
-                new AbstractModule() {
-                  @Provides
-                  OpenAIChatProcessorConfig provideOpenAIChatProcessorConfig() {
-                    return OpenAIChatProcessorConfig.getDefaultInstance();
-                  }
-
-                  @Provides
-                  ConversationMemory provideConversationMemory() {
-                    return ConversationMemory.geDefaultInstance();
-                  }
-                }));
+    LangtorchHubModuleRegistry registry = LangtorchHubModuleRegistry.create();
+    registry.registerOpenAiModule(
+        OpenAIKeyConfig.builder().setReadFromEnvFile(false).setOpenAiApiKey(openAIKey).build());
```

Comment:
I feel it's bit verbose that user needs to set two mutually exclusive fields.
I am not sure which one is considered better option:
1. Explicit key always has higher priority
2.  do validation inside build() method: if ReadFromEnvFile and OpenAiApiKey are both set, thrown an exception?

Reply:
thanks, agree with the point that pattern is verbose. Actually, I want one more step further. I'd like to disable the builder/setter to disallow user set both readFromEnv/ApiKey but only create specific type config, with the benefit of those factory methods we can skip option-check in the following functions.